rfd/0128-aws-integration-tests.md

```diff
@@ -0,0 +1,252 @@
+
+Teleport Auth and Proxy configurations are similar to the configurations we
+use for other tests. The discovery service will be configured to discover the healthy EKS
+clusters with `env: ci` tag. The Kubernetes service will be configured to
+watch clusters discovered by the discovery service with `env: ci` label.
+
+#### Discovery service config
+
+E2E tests will configure the Discovery service to discover EKS clusters with
+`env: ci` tag. The snippet below shows the configuration that will be used
+by the E2E tests.
+
+```yaml
+
+discovery_service:
+    enabled: "yes"
+    # discovery_group is used to group discovered resources into different
+    # sets. This is useful when you have multiple Teleport Discovery services
+    # running in the same cluster but polling different cloud providers or cloud
+    # accounts. It prevents discovered services from colliding in Teleport when
+    # managing discovered resources.
+    discovery_group: "ci"
+    aws:
+     - types: ["eks"]
+       # AWS regions to search for resources from
+       regions: ["us-east-1", "us-west-1"]
+       tags:
+         "env": "ci"
+
+```
+
+#### Kubernetes service config
+
+E2E tests will configure the Kubernetes service to watch `kube_cluster` objects
+with label `env:ci` and serve them. The snippet bellow shows the configuration that
+will be used.
+
+```yaml
+kubernetes_service:
+  enabled: "yes"
+  resources:
+  - tags:
+      "env": "ci"
+```
+
+Once the Teleport components are running, the test expects that the Kubernetes
+service starts heartbeat the clusters discovered by the discovery service.
+`tsh kube ls` will be used to monitor the clusters being served by the Kubernetes
+service. Once the cluster shows up in the list, we consider it ready.
+
+After the initial discovery phase, E2E tests will use `tsh` to simulate client connections
+to the cluster and run a simple command to ensure that the connection works and is forwarded as expected.
+`tsh kube login <kube_cluster>` will be used to generate the kubeconfig, and
+`tsh kubectl get service` will be used to test the connection to the cluster.
+
+#### Requirements
+
+This section describes the requirements for the E2E tests for Kubernetes
+access.
+
+- One or more EKS control plane clusters running on AWS. We don't need that the
+  clusters are running any workloads, we only need that their control plane is
+  running and accessible.
+- Discovery service configured to discover the EKS clusters and with List and Describe
+  permissions to the EKS API. [Permissions](https://goteleport.com/docs/kubernetes-access/discovery/aws/#step-13-set-up-aws-iam-credentials)
+- Kubernetes service IAM role configured with the minimum required permissions
+  to generate the short-lived token and forward the requests to the Kubernetes
+  API. [IAM Mapping](https://goteleport.com/docs/kubernetes-access/discovery/aws/#iam-mapping)
+
+Spin up the EKS clusters takes a long time and it's not feasible to do it for
```

Comment:
We have discussed this on the engineering offsite and one change to this system we'd like to make is instead of maintaining a set of persistent resources which can become obsolete, we want to recreate these on some cadence (e.g. daily) so we don't have to worry about patch management, updates, etc.

Can you please update the RFD to that effect? Basically, running "terraform destroy/apply" that recreates these on a daily basis.

Reply:
Sure. Will do once I am back

Reply:
if we run e2e tests on every PR, how will we handle this routine recreation of the test infrastructure?